### PR TITLE
Improve public API queries for releases when names_only is true

### PR DESCRIPTION
**Pull request by bhearsum**
Locally, this cut the time for `names_only=1` queries in half.

### CODE REVIEW

File: src/auslib/services/releases.py [hunk 1]
```diff
@@ -244,6 +244,14 @@ def get_assets(name, trans):
     return assets
 
 
+def get_release_names(trans, product=None):
```
Comment (escapewindow) [inline src/auslib/services/releases.py:247]: Hm, do we want to cache the result? I imagine the answer might be "no" if the DB changed, and "maybe" or "yes" if the DB hasn't, depending on how expensive it is.

**Review by escapewindow — APPROVED** [review]
I'm not terribly familiar with this, so up to you if you want a deeper code review.

File: src/auslib/services/releases.py (hunk 1 above)
Comment (bhearsum) [reply src/auslib/services/releases.py:247]: I thought about this, but I think ultimately it won't win much, if anything. We have so many webheads (and they recycle) and so few requests (once per nightly decision task AFAICT), that the hit rate will be zero, or close enough to it.

File: src/auslib/services/releases.py [hunk 2]
```diff
@@ -244,6 +244,14 @@ def get_assets(name, trans):
...
+    where = {}
+    if product:
+        where["product"] = product
+
+    return sorted([r["name"] for r in dbo.releases_json.select(columns=[dbo.releases_json.name], where=where)])
```
Comment (jcristau) [inline src/auslib/services/releases.py:252]: does this intentionally not pass `transaction=trans` down?

Comment (bhearsum) [reply src/auslib/services/releases.py:252]: No :). Thank your for catching that.

**Pull request merged**